org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/ArrayExtensions.java

```diff
@@ -367,6 +438,29 @@ public static boolean equals(float[] array, Object other) {
 		return array.clone();
 	}
 	
+	 /**
+     * Returns whether the array contains the given element.
+     * 
+     * More formally, returns <tt>true</tt> if and only if this array contains
+     * at least one element <tt>e</tt> such that
+     * <tt>(o==null&nbsp;?&nbsp;e==null&nbsp;:&nbsp;o.equals(e))</tt>.
+     *
+     * @param array
+     * 			the array to test
+     * @param element 
+     * 			element whose presence in this array is to be tested
+     * @return <tt>true</tt> if this array contains the specified element
+     */
+	@Pure
+	public static boolean contains(float[] array, float element) {
+		for (int i = 0; i < array.length; i++) {
```

Comment:
For float and double, please check the logic of `java.util.Arrays.binarySearch0(double[], int, int, double)` to properly deal with NaN value etc.

Reply:
for double and float I switched to  java.lang.Double.compare(double, double) resp. java.lang.Float.compare(float, float) to check for equality.

Reply:
Nice